odoo/tools/translate.py

```diff
@@ -259,18 +264,25 @@ def process(node):
             result.tail = node.tail
             has_text = (
                 todo_has_text or nonspace(result.text) or nonspace(result.tail)
-                or any((key in TRANSLATED_ATTRS and val) for key, val in result.attrib.items())
+                or any((key in TRANSLATED_ATTRS and val) for key, val in attrib)
             )
             return (has_text, result)
 
         # translate the content of todo and append it to result
         append_content(result, translate_content(todo) if todo_has_text else todo)
 
         # translate the required attributes
-        for name, value in result.attrib.items():
+        for name, value in attrib:
             if name in TRANSLATED_ATTRS:
                 result.set(name, translate_text(value) or value)
 
+        # translate the attributes on some unique border cases.
+        for xpath, attribute in TRANSLATED_XPATH_ATTRS:
+            if result.xpath(xpath):
+                for name, value in attrib:
```

Comment:
What is the point of iterating over all the attributes instead of just... getting the value for `attribute` and translating it if it's present?

Reply:
Probably none indeed, bad copy/paste from FJA I guess.